src/jdk.jcmd/share/classes/sun/tools/jmap/JMap.java

```diff
@@ -244,8 +251,17 @@ private static void dump(String pid, String options)
 
         System.out.flush();
 
+        // There is a limitation that at most 3 arguments could be passed to hotspot.
+        // For backward compatibility, pass filename and liveopt as 1st and 2nd argument and
+        // then compose remaining arguments as the 3rd one, and use comma to seperate them.
+        // The difinition of 3rd argument in current implementation is:
+        //        "compress_level,parallel"
+        // Not that making all arguments as a whole string like jcmd did does not guarantee the
+        // compatiabilty when the new jmap is uses on old version of JDK.
+        // See AttachOperation::arg_count_max in attachListener.hpp for argument count limitation.
+        String more_args = compress_level + "," + parallel;
         // dumpHeap is not the same as jcmd GC.heap_dump
```

Comment:
To-discuss: 
Compatibility issue that new version of jmap work on old JDK may wrongly processing the following case:
jmap -dump:file=dump.hprof,gz=1,parallel=2 <pid>
this will cause the gz option not recognized correctly.

Can not simply change arg_count_max to 4, because it could cause JDK hang on waiting for arguments when old jmap is used.